src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,116 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+import java.util.Objects;
+
 public class ArrayList<T> implements List<T> {
+    private static final int CAPACITY = 10;
+    private T[] list;
+    private int size;
+
+    ArrayList() {
+        list = (T[]) new Object[CAPACITY];
+        size = 0;
+    }
+
     @Override
     public void add(T value) {
-
+        add(value, size);
     }
 
     @Override
     public void add(T value, int index) {
-
+        grow();
+        if (index != size) {
+            indexCheck(index);
+            System.arraycopy(list, index, list, index + 1, size - index);
+        }
+        list[index] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
```

Comment:
Naming again, despite our code works well, names shoudln't be ambiguous. Please rename either ArrayList field or incoming parameter.

Reply:
This is absolutely necessary. In java it is the best practice to use interface instead of concrete implementation.